pyprophet/export_parquet.py

```diff
@@ -0,0 +1,302 @@
+
+        LEFT JOIN FEATURE ON FEATURE_TRANSITION.FEATURE_ID = FEATURE.ID
+        LEFT JOIN FEATURE_MS1 ON FEATURE_TRANSITION.FEATURE_ID = FEATURE_MS1.FEATURE_ID
+        LEFT JOIN FEATURE_MS2 ON FEATURE_TRANSITION.FEATURE_ID = FEATURE_MS2.FEATURE_ID
+
+        LEFT JOIN RUN ON FEATURE.RUN_ID = RUN.ID
+
+        LEFT JOIN SCORE_MS2 ON FEATURE.ID = SCORE_MS2.FEATURE_ID
+        LEFT JOIN SCORE_PEPTIDE ON PEPTIDE.ID = SCORE_PEPTIDE.PEPTIDE_ID
+        LEFT JOIN SCORE_PROTEIN ON PROTEIN.ID = SCORE_PROTEIN.PROTEIN_ID
+
+        WHERE FEATURE.ID IS NULL
+
+        '''.format(columnsToSelect)
+
+    else: # each row will be a precursor/feature
+        query = '''
+        SELECT {0}
+        FROM FEATURE
+
+        LEFT JOIN PRECURSOR ON FEATURE.PRECURSOR_ID = PRECURSOR.ID
+        LEFT JOIN PRECURSOR_PEPTIDE_MAPPING ON PRECURSOR.ID = PRECURSOR_PEPTIDE_MAPPING.PRECURSOR_ID
+        LEFT JOIN PEPTIDE ON PRECURSOR_PEPTIDE_MAPPING.PEPTIDE_ID = PEPTIDE.ID
+        LEFT JOIN PEPTIDE_PROTEIN_MAPPING ON PEPTIDE.ID = PEPTIDE_PROTEIN_MAPPING.PEPTIDE_ID
+        LEFT JOIN PROTEIN ON PEPTIDE_PROTEIN_MAPPING.PROTEIN_ID = PROTEIN.ID
+
+        LEFT JOIN RUN ON FEATURE.RUN_ID = RUN.ID
+        LEFT JOIN FEATURE_MS1 ON FEATURE.ID = FEATURE_MS1.FEATURE_ID
+        LEFT JOIN FEATURE_MS2 ON FEATURE.ID = FEATURE_MS2.FEATURE_ID
+
+
+        LEFT JOIN SCORE_MS2 ON FEATURE.ID = SCORE_MS2.FEATURE_ID
+        LEFT JOIN SCORE_PEPTIDE ON PEPTIDE.ID = SCORE_PEPTIDE.PEPTIDE_ID
+        LEFT JOIN SCORE_PROTEIN ON PROTEIN.ID = SCORE_PROTEIN.PROTEIN_ID
+
+
+        UNION
+
+        SELECT DISTINCT {0} FROM PRECURSOR
+
+        LEFT JOIN PRECURSOR_PEPTIDE_MAPPING ON PRECURSOR.ID = PRECURSOR_PEPTIDE_MAPPING.PRECURSOR_ID
+        LEFT JOIN PEPTIDE ON PRECURSOR_PEPTIDE_MAPPING.PEPTIDE_ID = PEPTIDE.ID
+        LEFT JOIN PEPTIDE_PROTEIN_MAPPING ON PEPTIDE.ID = PEPTIDE_PROTEIN_MAPPING.PEPTIDE_ID
+        LEFT JOIN PROTEIN ON PEPTIDE_PROTEIN_MAPPING.PROTEIN_ID = PROTEIN.ID
+
+        LEFT JOIN FEATURE ON FEATURE.PRECURSOR_ID = PRECURSOR.ID
+        LEFT JOIN FEATURE_MS1 ON FEATURE.ID = FEATURE_MS1.FEATURE_ID
+        LEFT JOIN FEATURE_MS2 ON FEATURE.ID = FEATURE_MS2.FEATURE_ID
+
+        LEFT JOIN RUN ON FEATURE.RUN_ID = RUN.ID
+
+        LEFT JOIN SCORE_MS2 ON FEATURE.ID = SCORE_MS2.FEATURE_ID
+        LEFT JOIN SCORE_PEPTIDE ON PEPTIDE.ID = SCORE_PEPTIDE.PEPTIDE_ID
+        LEFT JOIN SCORE_PROTEIN ON PROTEIN.ID = SCORE_PROTEIN.PROTEIN_ID
+
+        WHERE FEATURE.ID IS NULL
+
+        '''.format(columnsToSelect)
+
+    now = datetime.now()
+    current_time = now.strftime("%H:%M:%S")
+    click.echo("Info: Executing Query (Current Time = {})".format(current_time))
+    
+    try:
+        click.echo("Info: Attempting Execution")
+        df = pd.read_sql(query, con)
+        click.echo("Info: read_sql_query went well")
+    except Exception as e:
+        click.echo("Info: read_sql_query failed: "+ str(e))
```

Comment:
@jcharkow is there a reason this is in a try-except block? If pandas fails to read in the data from the sql query, then you probably don't want to have an exception, but raise and error. Otherwise it will continue, but then error out because df wouldn't be defined.

I actually run into this error when testing this module, because my machine runs out of disk space.

The sql query also seems to take a really long time (on a 11Gb 134 runs merged osw, proteomics data only, no ipf). Took ~ 2 hours and then failed because of the disk space issue.

```
Executing Query (Current Time = 19:45:55)
Attempting Execution

read_sql_query failed: Execution failed on sql '
        SELECT PRECURSOR.TRAML_ID as "PRECURSOR.TRAML_ID", PRECURSOR.GROUP_LABEL as "PRECURSOR.GROUP_LABEL", PRECURSOR.PRECURSOR_MZ as "PRECURSOR.PRECURSOR_MZ", PRECURSOR.CHARGE as "PRECURSOR.CHARGE", PRECURSOR.LIBRARY_INTENSITY as "PRECURSOR.LIBRARY_INTENSITY", PRECURSOR.LIBRARY_RT as "PRECURSOR.LIBRARY_RT", PRECURSOR.LIBRARY_DRIFT_TIME as "PRECURSOR.LIBRARY_DRIFT_TIME", PRECURSOR.DECOY as DECOY, PEPTIDE.UNMODIFIED_SEQUENCE as "PEPTIDE.UNMODIFIED_SEQUENCE", PEPTIDE.MODIFIED_SEQUENCE as "PEPTIDE.MODIFIED_SEQUENCE", PROTEIN.PROTEIN_ACCESSION as "PROTEIN.PROTEIN_ACCESSION", FEATURE.RUN_ID as "FEATURE.RUN_ID", FEATURE.EXP_RT as "FEATURE.EXP_RT", FEATURE.EXP_IM as "FEATURE.EXP_IM", FEATURE.NORM_RT as "FEATURE.NORM_RT", FEATURE.DELTA_RT as "FEATURE.DELTA_RT", FEATURE.LEFT_WIDTH as "FEATURE.LEFT_WIDTH", FEATURE.RIGHT_WIDTH as "FEATURE.RIGHT_WIDTH", FEATURE.ID as FEATURE_ID, FEATURE_MS2.AREA_INTENSITY as "FEATURE_MS2.AREA_INTENSITY", FEATURE_MS2.TOTAL_AREA_INTENSITY as "FEATURE_MS2.TOTAL_AREA_INTENSITY", FEATURE_MS2.APEX_INTENSITY as "FEATURE_MS2.APEX_INTENSITY", FEATURE_MS2.TOTAL_MI as "FEATURE_MS2.TOTAL_MI", FEATURE_MS2.VAR_BSERIES_SCORE as "FEATURE_MS2.VAR_BSERIES_SCORE", FEATURE_MS2.VAR_DOTPROD_SCORE as "FEATURE_MS2.VAR_DOTPROD_SCORE", FEATURE_MS2.VAR_INTENSITY_SCORE as "FEATURE_MS2.VAR_INTENSITY_SCORE", FEATURE_MS2.VAR_ISOTOPE_CORRELATION_SCORE as "FEATURE_MS2.VAR_ISOTOPE_CORRELATION_SCORE", FEATURE_MS2.VAR_ISOTOPE_OVERLAP_SCORE as "FEATURE_MS2.VAR_ISOTOPE_OVERLAP_SCORE", FEATURE_MS2.VAR_LIBRARY_CORR as "FEATURE_MS2.VAR_LIBRARY_CORR", FEATURE_MS2.VAR_LIBRARY_DOTPROD as "FEATURE_MS2.VAR_LIBRARY_DOTPROD", FEATURE_MS2.VAR_LIBRARY_MANHATTAN as "FEATURE_MS2.VAR_LIBRARY_MANHATTAN", FEATURE_MS2.VAR_LIBRARY_RMSD as "FEATURE_MS2.VAR_LIBRARY_RMSD", FEATURE_MS2.VAR_LIBRARY_ROOTMEANSQUARE as "FEATURE_MS2.VAR_LIBRARY_ROOTMEANSQUARE", FEATURE_MS2.VAR_LIBRARY_SANGLE as "FEATURE_MS2.VAR_LIBRARY_SANGLE", FEATURE_MS2.VAR_LOG_SN_SCORE as "FEATURE_MS2.VAR_LOG_SN_SCORE", FEATURE_MS2.VAR_MANHATTAN_SCORE as "FEATURE_MS2.VAR_MANHATTAN_SCORE", FEATURE_MS2.VAR_MASSDEV_SCORE as "FEATURE_MS2.VAR_MASSDEV_SCORE", FEATURE_MS2.VAR_MASSDEV_SCORE_WEIGHTED as "FEATURE_MS2.VAR_MASSDEV_SCORE_WEIGHTED", FEATURE_MS2.VAR_MI_SCORE as "FEATURE_MS2.VAR_MI_SCORE", FEATURE_MS2.VAR_MI_WEIGHTED_SCORE as "FEATURE_MS2.VAR_MI_WEIGHTED_SCORE", FEATURE_MS2.VAR_MI_RATIO_SCORE as "FEATURE_MS2.VAR_MI_RATIO_SCORE", FEATURE_MS2.VAR_NORM_RT_SCORE as "FEATURE_MS2.VAR_NORM_RT_SCORE", FEATURE_MS2.VAR_XCORR_COELUTION as "FEATURE_MS2.VAR_XCORR_COELUTION", FEATURE_MS2.VAR_XCORR_COELUTION_WEIGHTED as "FEATURE_MS2.VAR_XCORR_COELUTION_WEIGHTED", FEATURE_MS2.VAR_XCORR_SHAPE as "FEATURE_MS2.VAR_XCORR_SHAPE", FEATURE_MS2.VAR_XCORR_SHAPE_WEIGHTED as "FEATURE_MS2.VAR_XCORR_SHAPE_WEIGHTED", FEATURE_MS2.VAR_YSERIES_SCORE as "FEATURE_MS2.VAR_YSERIES_SCORE", FEATURE_MS2.VAR_ELUTION_MODEL_FIT_SCORE as "FEATURE_MS2.VAR_ELUTION_MODEL_FIT_SCORE", FEATURE_MS2.VAR_IM_XCORR_SHAPE as "FEATURE_MS2.VAR_IM_XCORR_SHAPE", FEATURE_MS2.VAR_IM_XCORR_COELUTION as "FEATURE_MS2.VAR_IM_XCORR_COELUTION", FEATURE_MS2.VAR_IM_DELTA_SCORE as "FEATURE_MS2.VAR_IM_DELTA_SCORE", FEATURE_MS2.VAR_SONAR_LAG as "FEATURE_MS2.VAR_SONAR_LAG", FEATURE_MS2.VAR_SONAR_SHAPE as "FEATURE_MS2.VAR_SONAR_SHAPE", FEATURE_MS2.VAR_SONAR_LOG_SN as "FEATURE_MS2.VAR_SONAR_LOG_SN", FEATURE_MS2.VAR_SONAR_LOG_DIFF as "FEATURE_MS2.VAR_SONAR_LOG_DIFF", FEATURE_MS2.VAR_SONAR_LOG_TREND as "FEATURE_MS2.VAR_SONAR_LOG_TREND", FEATURE_MS2.VAR_SONAR_RSQ as "FEATURE_MS2.VAR_SONAR_RSQ", FEATURE_MS1.AREA_INTENSITY as "FEATURE_MS1.AREA_INTENSITY", FEATURE_MS1.APEX_INTENSITY as "FEATURE_MS1.APEX_INTENSITY", FEATURE_MS1.VAR_MASSDEV_SCORE as "FEATURE_MS1.VAR_MASSDEV_SCORE", FEATURE_MS1.VAR_MI_SCORE as "FEATURE_MS1.VAR_MI_SCORE", FEATURE_MS1.VAR_MI_CONTRAST_SCORE as "FEATURE_MS1.VAR_MI_CONTRAST_SCORE", FEATURE_MS1.VAR_MI_COMBINED_SCORE as "FEATURE_MS1.VAR_MI_COMBINED_SCORE", FEATURE_MS1.VAR_ISOTOPE_CORRELATION_SCORE as "FEATURE_MS1.VAR_ISOTOPE_CORRELATION_SCORE", FEATURE_MS1.VAR_ISOTOPE_OVERLAP_SCORE as "FEATURE_MS1.VAR_ISOTOPE_OVERLAP_SCORE", FEATURE_MS1.VAR_IM_MS1_DELTA_SCORE as "FEATURE_MS1.VAR_IM_MS1_DELTA_SCORE", FEATURE_MS1.VAR_XCORR_COELUTION as "FEATURE_MS1.VAR_XCORR_COELUTION", FEATURE_MS1.VAR_XCORR_COELUTION_CONTRAST as "FEATURE_MS1.VAR_XCORR_COELUTION_CONTRAST", FEATURE_MS1.VAR_XCORR_COELUTION_COMBINED as "FEATURE_MS1.VAR_XCORR_COELUTION_COMBINED", FEATURE_MS1.VAR_XCORR_SHAPE as "FEATURE_MS1.VAR_XCORR_SHAPE", FEATURE_MS1.VAR_XCORR_SHAPE_CONTRAST as "FEATURE_MS1.VAR_XCORR_SHAPE_CONTRAST", FEATURE_MS1.VAR_XCORR_SHAPE_COMBINED as "FEATURE_MS1.VAR_XCORR_SHAPE_COMBINED", SCORE_MS2.SCORE as "SCORE_MS2.SCORE", SCORE_MS2.RANK as "SCORE_MS2.RANK", SCORE_MS2.PVALUE as "SCORE_MS2.PVALUE", SCORE_MS2.QVALUE as "SCORE_MS2.QVALUE", SCORE_MS2.PEP as "SCORE_MS2.PEP", SCORE_PEPTIDE.CONTEXT as "SCORE_PEPTIDE.CONTEXT", SCORE_PEPTIDE.SCORE as "SCORE_PEPTIDE.SCORE", SCORE_PEPTIDE.PVALUE as "SCORE_PEPTIDE.PVALUE", SCORE_PEPTIDE.QVALUE as "SCORE_PEPTIDE.QVALUE", SCORE_PEPTIDE.PEP as "SCORE_PEPTIDE.PEP", SCORE_PROTEIN.SCORE as "SCORE_PROTEIN.SCORE", SCORE_PROTEIN.PVALUE as "SCORE_PROTEIN.PVALUE", SCORE_PROTEIN.QVALUE as "SCORE_PROTEIN.QVALUE", SCORE_PROTEIN.PEP as "SCORE_PROTEIN.PEP", RUN.FILENAME as "RUN.FILENAME", PRECURSOR_PEPTIDE_MAPPING.PEPTIDE_ID as PEPTIDE_ID, PRECURSOR_PEPTIDE_MAPPING.PRECURSOR_ID as PRECURSOR_ID, PEPTIDE_PROTEIN_MAPPING.PROTEIN_ID as PROTEIN_ID
        FROM FEATURE

        LEFT JOIN PRECURSOR ON FEATURE.PRECURSOR_ID = PRECURSOR.ID
        LEFT JOIN PRECURSOR_PEPTIDE_MAPPING ON PRECURSOR.ID = PRECURSOR_PEPTIDE_MAPPING.PRECURSOR_ID
        LEFT JOIN PEPTIDE ON PRECURSOR_PEPTIDE_MAPPING.PEPTIDE_ID = PEPTIDE.ID
        LEFT JOIN PEPTIDE_PROTEIN_MAPPING ON PEPTIDE.ID = PEPTIDE_PROTEIN_MAPPING.PEPTIDE_ID
        LEFT JOIN PROTEIN ON PEPTIDE_PROTEIN_MAPPING.PROTEIN_ID = PROTEIN.ID

        LEFT JOIN RUN ON FEATURE.RUN_ID = RUN.ID
        LEFT JOIN FEATURE_MS1 ON FEATURE.ID = FEATURE_MS1.FEATURE_ID
        LEFT JOIN FEATURE_MS2 ON FEATURE.ID = FEATURE_MS2.FEATURE_ID


        LEFT JOIN SCORE_MS2 ON FEATURE.ID = SCORE_MS2.FEATURE_ID
        LEFT JOIN SCORE_PEPTIDE ON PEPTIDE.ID = SCORE_PEPTIDE.PEPTIDE_ID
        LEFT JOIN SCORE_PROTEIN ON PROTEIN.ID = SCORE_PROTEIN.PROTEIN_ID


        UNION

        SELECT DISTINCT PRECURSOR.TRAML_ID as "PRECURSOR.TRAML_ID", PRECURSOR.GROUP_LABEL as "PRECURSOR.GROUP_LABEL", PRECURSOR.PRECURSOR_MZ as "PRECURSOR.PRECURSOR_MZ", PRECURSOR.CHARGE as "PRECURSOR.CHARGE", PRECURSOR.LIBRARY_INTENSITY as "PRECURSOR.LIBRARY_INTENSITY", PRECURSOR.LIBRARY_RT as "PRECURSOR.LIBRARY_RT", PRECURSOR.LIBRARY_DRIFT_TIME as "PRECURSOR.LIBRARY_DRIFT_TIME", PRECURSOR.DECOY as DECOY, PEPTIDE.UNMODIFIED_SEQUENCE as "PEPTIDE.UNMODIFIED_SEQUENCE", PEPTIDE.MODIFIED_SEQUENCE as "PEPTIDE.MODIFIED_SEQUENCE", PROTEIN.PROTEIN_ACCESSION as "PROTEIN.PROTEIN_ACCESSION", FEATURE.RUN_ID as "FEATURE.RUN_ID", FEATURE.EXP_RT as "FEATURE.EXP_RT", FEATURE.EXP_IM as "FEATURE.EXP_IM", FEATURE.NORM_RT as "FEATURE.NORM_RT", FEATURE.DELTA_RT as "FEATURE.DELTA_RT", FEATURE.LEFT_WIDTH as "FEATURE.LEFT_WIDTH", FEATURE.RIGHT_WIDTH as "FEATURE.RIGHT_WIDTH", FEATURE.ID as FEATURE_ID, FEATURE_MS2.AREA_INTENSITY as "FEATURE_MS2.AREA_INTENSITY", FEATURE_MS2.TOTAL_AREA_INTENSITY as "FEATURE_MS2.TOTAL_AREA_INTENSITY", FEATURE_MS2.APEX_INTENSITY as "FEATURE_MS2.APEX_INTENSITY", FEATURE_MS2.TOTAL_MI as "FEATURE_MS2.TOTAL_MI", FEATURE_MS2.VAR_BSERIES_SCORE as "FEATURE_MS2.VAR_BSERIES_SCORE", FEATURE_MS2.VAR_DOTPROD_SCORE as "FEATURE_MS2.VAR_DOTPROD_SCORE", FEATURE_MS2.VAR_INTENSITY_SCORE as "FEATURE_MS2.VAR_INTENSITY_SCORE", FEATURE_MS2.VAR_ISOTOPE_CORRELATION_SCORE as "FEATURE_MS2.VAR_ISOTOPE_CORRELATION_SCORE", FEATURE_MS2.VAR_ISOTOPE_OVERLAP_SCORE as "FEATURE_MS2.VAR_ISOTOPE_OVERLAP_SCORE", FEATURE_MS2.VAR_LIBRARY_CORR as "FEATURE_MS2.VAR_LIBRARY_CORR", FEATURE_MS2.VAR_LIBRARY_DOTPROD as "FEATURE_MS2.VAR_LIBRARY_DOTPROD", FEATURE_MS2.VAR_LIBRARY_MANHATTAN as "FEATURE_MS2.VAR_LIBRARY_MANHATTAN", FEATURE_MS2.VAR_LIBRARY_RMSD as "FEATURE_MS2.VAR_LIBRARY_RMSD", FEATURE_MS2.VAR_LIBRARY_ROOTMEANSQUARE as "FEATURE_MS2.VAR_LIBRARY_ROOTMEANSQUARE", FEATURE_MS2.VAR_LIBRARY_SANGLE as "FEATURE_MS2.VAR_LIBRARY_SANGLE", FEATURE_MS2.VAR_LOG_SN_SCORE as "FEATURE_MS2.VAR_LOG_SN_SCORE", FEATURE_MS2.VAR_MANHATTAN_SCORE as "FEATURE_MS2.VAR_MANHATTAN_SCORE", FEATURE_MS2.VAR_MASSDEV_SCORE as "FEATURE_MS2.VAR_MASSDEV_SCORE", FEATURE_MS2.VAR_MASSDEV_SCORE_WEIGHTED as "FEATURE_MS2.VAR_MASSDEV_SCORE_WEIGHTED", FEATURE_MS2.VAR_MI_SCORE as "FEATURE_MS2.VAR_MI_SCORE", FEATURE_MS2.VAR_MI_WEIGHTED_SCORE as "FEATURE_MS2.VAR_MI_WEIGHTED_SCORE", FEATURE_MS2.VAR_MI_RATIO_SCORE as "FEATURE_MS2.VAR_MI_RATIO_SCORE", FEATURE_MS2.VAR_NORM_RT_SCORE as "FEATURE_MS2.VAR_NORM_RT_SCORE", FEATURE_MS2.VAR_XCORR_COELUTION as "FEATURE_MS2.VAR_XCORR_COELUTION", FEATURE_MS2.VAR_XCORR_COELUTION_WEIGHTED as "FEATURE_MS2.VAR_XCORR_COELUTION_WEIGHTED", FEATURE_MS2.VAR_XCORR_SHAPE as "FEATURE_MS2.VAR_XCORR_SHAPE", FEATURE_MS2.VAR_XCORR_SHAPE_WEIGHTED as "FEATURE_MS2.VAR_XCORR_SHAPE_WEIGHTED", FEATURE_MS2.VAR_YSERIES_SCORE as "FEATURE_MS2.VAR_YSERIES_SCORE", FEATURE_MS2.VAR_ELUTION_MODEL_FIT_SCORE as "FEATURE_MS2.VAR_ELUTION_MODEL_FIT_SCORE", FEATURE_MS2.VAR_IM_XCORR_SHAPE as "FEATURE_MS2.VAR_IM_XCORR_SHAPE", FEATURE_MS2.VAR_IM_XCORR_COELUTION as "FEATURE_MS2.VAR_IM_XCORR_COELUTION", FEATURE_MS2.VAR_IM_DELTA_SCORE as "FEATURE_MS2.VAR_IM_DELTA_SCORE", FEATURE_MS2.VAR_SONAR_LAG as "FEATURE_MS2.VAR_SONAR_LAG", FEATURE_MS2.VAR_SONAR_SHAPE as "FEATURE_MS2.VAR_SONAR_SHAPE", FEATURE_MS2.VAR_SONAR_LOG_SN as "FEATURE_MS2.VAR_SONAR_LOG_SN", FEATURE_MS2.VAR_SONAR_LOG_DIFF as "FEATURE_MS2.VAR_SONAR_LOG_DIFF", FEATURE_MS2.VAR_SONAR_LOG_TREND as "FEATURE_MS2.VAR_SONAR_LOG_TREND", FEATURE_MS2.VAR_SONAR_RSQ as "FEATURE_MS2.VAR_SONAR_RSQ", FEATURE_MS1.AREA_INTENSITY as "FEATURE_MS1.AREA_INTENSITY", FEATURE_MS1.APEX_INTENSITY as "FEATURE_MS1.APEX_INTENSITY", FEATURE_MS1.VAR_MASSDEV_SCORE as "FEATURE_MS1.VAR_MASSDEV_SCORE", FEATURE_MS1.VAR_MI_SCORE as "FEATURE_MS1.VAR_MI_SCORE", FEATURE_MS1.VAR_MI_CONTRAST_SCORE as "FEATURE_MS1.VAR_MI_CONTRAST_SCORE", FEATURE_MS1.VAR_MI_COMBINED_SCORE as "FEATURE_MS1.VAR_MI_COMBINED_SCORE", FEATURE_MS1.VAR_ISOTOPE_CORRELATION_SCORE as "FEATURE_MS1.VAR_ISOTOPE_CORRELATION_SCORE", FEATURE_MS1.VAR_ISOTOPE_OVERLAP_SCORE as "FEATURE_MS1.VAR_ISOTOPE_OVERLAP_SCORE", FEATURE_MS1.VAR_IM_MS1_DELTA_SCORE as "FEATURE_MS1.VAR_IM_MS1_DELTA_SCORE", FEATURE_MS1.VAR_XCORR_COELUTION as "FEATURE_MS1.VAR_XCORR_COELUTION", FEATURE_MS1.VAR_XCORR_COELUTION_CONTRAST as "FEATURE_MS1.VAR_XCORR_COELUTION_CONTRAST", FEATURE_MS1.VAR_XCORR_COELUTION_COMBINED as "FEATURE_MS1.VAR_XCORR_COELUTION_COMBINED", FEATURE_MS1.VAR_XCORR_SHAPE as "FEATURE_MS1.VAR_XCORR_SHAPE", FEATURE_MS1.VAR_XCORR_SHAPE_CONTRAST as "FEATURE_MS1.VAR_XCORR_SHAPE_CONTRAST", FEATURE_MS1.VAR_XCORR_SHAPE_COMBINED as "FEATURE_MS1.VAR_XCORR_SHAPE_COMBINED", SCORE_MS2.SCORE as "SCORE_MS2.SCORE", SCORE_MS2.RANK as "SCORE_MS2.RANK", SCORE_MS2.PVALUE as "SCORE_MS2.PVALUE", SCORE_MS2.QVALUE as "SCORE_MS2.QVALUE", SCORE_MS2.PEP as "SCORE_MS2.PEP", SCORE_PEPTIDE.CONTEXT as "SCORE_PEPTIDE.CONTEXT", SCORE_PEPTIDE.SCORE as "SCORE_PEPTIDE.SCORE", SCORE_PEPTIDE.PVALUE as "SCORE_PEPTIDE.PVALUE", SCORE_PEPTIDE.QVALUE as "SCORE_PEPTIDE.QVALUE", SCORE_PEPTIDE.PEP as "SCORE_PEPTIDE.PEP", SCORE_PROTEIN.SCORE as "SCORE_PROTEIN.SCORE", SCORE_PROTEIN.PVALUE as "SCORE_PROTEIN.PVALUE", SCORE_PROTEIN.QVALUE as "SCORE_PROTEIN.QVALUE", SCORE_PROTEIN.PEP as "SCORE_PROTEIN.PEP", RUN.FILENAME as "RUN.FILENAME", PRECURSOR_PEPTIDE_MAPPING.PEPTIDE_ID as PEPTIDE_ID, PRECURSOR_PEPTIDE_MAPPING.PRECURSOR_ID as PRECURSOR_ID, PEPTIDE_PROTEIN_MAPPING.PROTEIN_ID as PROTEIN_ID FROM PRECURSOR

        LEFT JOIN PRECURSOR_PEPTIDE_MAPPING ON PRECURSOR.ID = PRECURSOR_PEPTIDE_MAPPING.PRECURSOR_ID
        LEFT JOIN PEPTIDE ON PRECURSOR_PEPTIDE_MAPPING.PEPTIDE_ID = PEPTIDE.ID
        LEFT JOIN PEPTIDE_PROTEIN_MAPPING ON PEPTIDE.ID = PEPTIDE_PROTEIN_MAPPING.PEPTIDE_ID
        LEFT JOIN PROTEIN ON PEPTIDE_PROTEIN_MAPPING.PROTEIN_ID = PROTEIN.ID

        LEFT JOIN FEATURE ON FEATURE.PRECURSOR_ID = PRECURSOR.ID
        LEFT JOIN FEATURE_MS1 ON FEATURE.ID = FEATURE_MS1.FEATURE_ID
        LEFT JOIN FEATURE_MS2 ON FEATURE.ID = FEATURE_MS2.FEATURE_ID

        LEFT JOIN RUN ON FEATURE.RUN_ID = RUN.ID

        LEFT JOIN SCORE_MS2 ON FEATURE.ID = SCORE_MS2.FEATURE_ID
        LEFT JOIN SCORE_PEPTIDE ON PEPTIDE.ID = SCORE_PEPTIDE.PEPTIDE_ID
        LEFT JOIN SCORE_PROTEIN ON PROTEIN.ID = SCORE_PROTEIN.PROTEIN_ID

        WHERE FEATURE.ID IS NULL

        ': database or disk is full
Done Executing (Current Time = 21:53:18)
Creating bitwise maps ...
Traceback (most recent call last):
  File "/home/justincsing/anaconda3/envs/pytest39/bin/pyprophet", line 33, in <module>
    sys.exit(load_entry_point('pyprophet', 'console_scripts', 'pyprophet')())
  File "/home/justincsing/anaconda3/envs/pytest39/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/justincsing/anaconda3/envs/pytest39/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/justincsing/anaconda3/envs/pytest39/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1688, in invoke
    rv.append(sub_ctx.command.invoke(sub_ctx))
  File "/home/justincsing/anaconda3/envs/pytest39/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/justincsing/anaconda3/envs/pytest39/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/justincsing/anaconda3/envs/pytest39/lib/python3.9/site-packages/pyprophet-2.2.3-py3.9-linux-x86_64.egg/pyprophet/main.py", line 369, in export_parquet
  File "/home/justincsing/anaconda3/envs/pytest39/lib/python3.9/site-packages/pyprophet-2.2.3-py3.9-linux-x86_64.egg/pyprophet/export_parquet.py", line 251, in export_to_parquet
UnboundLocalError: local variable 'df' referenced before assignment
```

Reply:
No specific reason in a try/except clause except for debugging purposes and seeing how long it tries before failing. The memory issue is something that I've struggled with as well. Currently I've been running it on machines with large amounts of memory to avoid this issue.

Reply:
Do you have any suggestions for limiting memory consumption?

Reply:
I was think either breaking up the sql query or read/writing the data in chunks. Is it possible to append to a parquet file with pyarrow?